src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/SplashScreen.cs

```diff
@@ -62,25 +62,20 @@ public void Show(bool autoClose)
             Show(autoClose, false);
         }
 
-        public void Show(bool autoClose, bool topMost)
+        public unsafe void Show(bool autoClose, bool topMost)
         {
             // If we've already been shown it isn't an error to call show
             // again (maybe you forgot) since you will still be shown state.
-            if (_hwnd == IntPtr.Zero)
+            if (_hwnd.IsNull)
             {
                 UnmanagedMemoryStream umemStream;
                 using (umemStream = GetResourceStream())
                 {
                     if (umemStream != null)
                     {
                         umemStream.Seek(0, SeekOrigin.Begin); // ensure stream position
-                        IntPtr pImageSrcBuffer;
-                        unsafe
-                        {
-                            pImageSrcBuffer = new IntPtr(umemStream.PositionPointer);
-                        }
 
-                        if (CreateLayeredWindowFromImgBuffer(pImageSrcBuffer, umemStream.Length, topMost) && autoClose == true)
+                        if (CreateLayeredWindowFromImgBuffer(new(umemStream.PositionPointer, (int)umemStream.Length), topMost) && autoClose == true)
```

Comment:
💡 Explicit type would help here

Reply:
I don't think it has significant value here, but if the WPF team does that's fine.